plugins/lsp/default.nix

```diff
@@ -245,18 +245,17 @@ lib.nixvim.plugins.mkNeovimPlugin {
           }"}
 
           for i, server in ipairs(__lspServers) do
-            if type(server) == "string" then
-              require("lspconfig")[server].setup(__setup)
-            else
-              local options = ${runWrappers cfg.setupWrappers "server.extraOptions"}
-
-              if options == nil then
-                options = __setup
-              else
-                options = vim.tbl_extend("keep", options, __setup)
-              end
+            local name = server
+            local options = nil
+            if type(name) ~= 'string' then
+              name = server.name
+              options = server.extraOptions
+            end
```

Comment:
Nit: the earlier error with `server` being a string was not caught by the test suite. Is it possible to add a test-case for it being defined as a string?

Or maybe the option-type doesn't actually support it being a string, in which case this impl can be simplified?